src/App.jsx

```diff
@@ -284,7 +287,25 @@ const MainContent = ({
   page,
   setPage,
   scrollTarget,
+  totalElements,
 }) => {
+  /**
+   * function to calculate and show correct results for relevant page.
+   *
+   * @param {number} resultsLength - the amount of results per page.
+   *
+   * @return {string} - returns a string with the amount of results
+   */
+  const amountOfResults = resultsLength => {
```

Comment:
To make this even more reusable (and testable!) I would also suggest handing it all the info it needs through parameters. Looking at the method I have no way of knowing what `page` refers to is it total number of pages in the results or the current page number?

It's a bit unconventional that results is the number shown on that page already, usually you'd have

* what is the current page
* how many to show on the page
* total number of results

and you would work it out from that. I would suggest sticking to this formula to make it as reusable as possible!



```js
export default function App() {

  function pagination(numberOfResults, currentPage, itemsPerPage) {
    return {
      total: numberOfResults,
      per_page: itemsPerPage,
      current_page: currentPage,
      last_page: Math.ceil(numberOfResults / itemsPerPage),
      from: (currentPage - 1) * itemsPerPage + 1,
      to:
        currentPage * itemsPerPage < numberOfResults
          ? currentPage * itemsPerPage
          : numberOfResults
    };
  }

  const results = [...Array(100)].map((x) => []);
  const numberOfResults = results.length;
  const currentPage = 5;
  const itemsPerPage = 20;

  const { total, per_page, current_page, last_page, from, to } = pagination(
    numberOfResults,
    currentPage,
    itemsPerPage
  );

  return (
    <div className="App">
      {current_page <= last_page &&
        <p>Viewing {from} - {to} out of {total}</p>
      }
      <p>
      <br />
      Showing {per_page} results per page
      <br />
      You are on page number: {current_page}
      <br />
      The last page is number: {last_page} 
      </p>
    </div>
  );
}
```

With something like the above you can standardise your application.

Further up the page we have 

```js
      setTotalPages(data.totalPages)
      setTotalElements(data.totalElements)
```

These could be replaced with one `setPagination()` method for the above - we only use those states on search at the moment.